src/transformers/models/t5/modeling_t5.py

```diff
@@ -426,11 +441,14 @@ def _relative_position_bucket(relative_position, bidirectional=True, num_buckets
         relative_buckets += torch.where(is_small, relative_position, relative_position_if_large)
         return relative_buckets
 
-    def compute_bias(self, query_length, key_length, device=None):
+    def compute_bias(self, query_length, key_length, device=None, cache_position=None):
         """Compute binned relative position bias"""
         if device is None:
             device = self.relative_attention_bias.weight.device
-        context_position = torch.arange(query_length, dtype=torch.long, device=device)[:, None]
+        if cache_position is None:
+            context_position = torch.arange(query_length, dtype=torch.long, device=device)[:, None]
```

Comment:
just a not, this will not produce the correct generation when you have cudagraphs AFAIK! But it's good for the other cases (BC and eager modes)

Reply:
hmm, why not? Cache position should be same, as the prev context length neither accounts for pad tokens